Package.swift

```diff
@@ -3,6 +3,27 @@
 
 import PackageDescription
 import class Foundation.ProcessInfo
+import struct Foundation.URL
+
+// Extract compiler flags from Local.xcconfig, if any.
+var additionalCompilerFlags: [PackageDescription.SwiftSetting] = {
+    guard let config = try? String(
+        contentsOf: URL(fileURLWithPath: #file)
+            .deletingLastPathComponent()
+            .appendingPathComponent("Local.xcconfig")
+    ) else {
+        return []
+    }
+    // We split the capture group by space and remove any special flags, such as $(inherited).
+    return config
+        .firstMatch(of: #/^SWIFT_ACTIVE_COMPILATION_CONDITIONS *= *(.*)$/#.anchorsMatchLineEndings())?
+        .output
+        .1
+        .split(whereSeparator: \.isWhitespace)
+        .filter { !$0.isEmpty && !$0.hasPrefix("$") }
+        .map { .define(String($0)) }
+        ?? []
+}()
```

Comment:
This change will go out to users of our SDK too, right? 🤔 So if they have a `Local.xcconfig` file, this logic will get executed?

If that's the case, should we end up naming our file something more like `RCLocal.xcconfig` so it has less likely chance t pick anything else up? 🤷‍♂️

Reply:
That's a good concern! But I think we should be good. The `#file` macro [returns the path of the file in which it is defined](https://developer.apple.com/documentation/swift/file()). So for library consumers that is `/path/to/purchases-ios-spm/Package.swift`. Unless they have a `Local.xcconfig` file in the `purchases-ios-spm` folder (unlikely imo), nothing should get picked up. Let me know what you think! 

(I changed it to the more future-proof `#filePath` in https://github.com/RevenueCat/purchases-ios/pull/4368/commits/030a01554010dcb3725673a3d52fb91769ed1242.)